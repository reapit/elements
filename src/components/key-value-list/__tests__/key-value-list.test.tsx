import { render } from '@testing-library/react'
import { KeyValueList } from '..'

describe('KeyValueList component', () => {
  it('should match a snapshot', () => {
    const wrapper = render(
      <KeyValueList
        items={[
          {
            key: 'Name',
            value: 'John Doe',
            iconName: 'user',
            intent: 'primary',
          },
          {
            key: 'Email',
            value: 'email@example.com',
            iconName: 'email',
            intent: 'primary',
          },
          {
            key: 'Parking Spaces',
            value: '2',
            iconName: 'car',
            intent: 'primary',
          },
          {
            key: 'Date Signed Up',
            value: '20th September 2023',
            iconName: 'calendar',
            intent: 'primary',
          },
          {
            key: 'Address',
            value:
              'I added a textEllipsis prop to this one to show how it works. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.',
            iconName: 'property',
            intent: 'primary',
            textEllipsis: true,
          },
          {
            key: 'Description',
            value:
              'This one I just allowed the text to wrap. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.',
            iconName: 'edit',
            intent: 'primary',
          },
        ]}
      />,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot for a grid', () => {
    const wrapper = render(
      <KeyValueList
        hasGrid
        items={[
          {
            key: 'Name',
            value: 'John Doe',
            iconName: 'user',
            intent: 'primary',
            colSize: 'half',
          },
          {
            key: 'Email',
            value: 'email@example.com',
            iconName: 'email',
            intent: 'primary',
            colSize: 'half',
          },
          {
            key: 'Parking Spaces',
            value: '2',
            iconName: 'car',
            intent: 'primary',
            colSize: 'half',
          },
          {
            key: 'Date Signed Up',
            value: '20th September 2023',
            iconName: 'calendar',
            intent: 'primary',
            colSize: 'half',
          },
          {
            key: 'Address',
            value:
              'I added a textEllipsis prop to this one to show how it works. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.',
            iconName: 'property',
            intent: 'primary',
            colSize: 'full',
            textEllipsis: true,
          },
          {
            key: 'Description',
            value:
              'This one I just allowed the text to wrap. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.',
            iconName: 'edit',
            intent: 'primary',
            colSize: 'full',
          },
        ]}
      />,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
