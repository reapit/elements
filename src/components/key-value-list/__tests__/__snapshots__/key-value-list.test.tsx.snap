// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyValueList component > should match a snapshot 1`] = `
<DocumentFragment>
  <div
    role="list"
  >
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name user"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8.37998 8.88C8.37998 6.87911 9.99911 5.25551 12 5.25C14.0025 5.25549 15.6245 6.87749 15.63 8.88C15.6272 9.49712 15.5703 10.1128 15.46 10.72C15.3083 12.534 13.8195 13.9453 12 14C10.1805 13.9453 8.69171 12.534 8.54 10.72C8.43267 10.1126 8.37913 9.49685 8.37998 8.88ZM12 12C12.91 12 13.36 11.18 13.49 10.4C13.5796 9.89821 13.6265 9.38972 13.63 8.88C13.678 8.26901 13.3792 7.68276 12.8566 7.36257C12.334 7.04238 11.676 7.04238 11.1534 7.36257C10.6308 7.68276 10.332 8.26901 10.38 8.88C10.3789 9.38949 10.4224 9.8981 10.51 10.4C10.64 11.18 11.09 12 12 12Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M1 12C1 5.92487 5.92487 1 12 1C14.9174 1 17.7153 2.15893 19.7782 4.22183C21.8411 6.28473 23 9.08262 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 17C9.25 17 7.27 18.27 7 19.49H7.06C10.0671 21.5044 13.9929 21.5044 17 19.49C16.73 18.27 14.75 17 12 17ZM12 15C15.07 15 17.62 16.28 18.55 18.12H18.58C21.9161 14.545 21.7874 8.9604 18.2902 5.54284C14.793 2.12528 9.20695 2.12528 5.70975 5.54284C2.21255 8.9604 2.0839 14.545 5.42 18.12C6.38 16.28 8.93 15 12 15Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Name
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          John Doe
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name email"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 4H4C2.89543 4 2 4.89543 2 6V18C2 19.1046 2.89543 20 4 20H20C21.1046 20 22 19.1046 22 18V6C22 4.89543 21.1046 4 20 4ZM18.59 6L12 12.59L5.41 6H18.59ZM4 18V7.41L10.59 14C11.3701 14.7754 12.6299 14.7754 13.41 14L20 7.41V18H4Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Email
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          email@example.com
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name car"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.5 15.5C7.60457 15.5 8.5 14.6046 8.5 13.5C8.5 12.3954 7.60457 11.5 6.5 11.5C5.39543 11.5 4.5 12.3954 4.5 13.5C4.5 14.6046 5.39543 15.5 6.5 15.5Z"
              fill="currentColor"
            />
            <path
              d="M19.5 13.5C19.5 14.6046 18.6046 15.5 17.5 15.5C16.3954 15.5 15.5 14.6046 15.5 13.5C15.5 12.3954 16.3954 11.5 17.5 11.5C18.6046 11.5 19.5 12.3954 19.5 13.5Z"
              fill="currentColor"
            />
            <path
              clip-rule="evenodd"
              d="M21 8H20.72L19.63 4.73C19.0843 3.10029 17.5586 2.0013 15.84 2H8.16C6.43872 2.00274 4.9122 3.10634 4.37 4.74L3.28 8H3C1.89543 8 1 8.89543 1 10V17C1 18.1046 1.89543 19 3 19V20C3 21.1046 3.89543 22 5 22H6C7.10457 22 8 21.1046 8 20V19H16V20C16 21.1046 16.8954 22 18 22H19C20.1046 22 21 21.1046 21 20V19C22.1046 19 23 18.1046 23 17V10C23 8.89543 22.1046 8 21 8ZM6.26 5.37C6.53168 4.55142 7.29752 3.99922 8.16 4H15.84C16.7025 3.99922 17.4683 4.55142 17.74 5.37L18.61 8H5.39L6.26 5.37ZM6 20H5V19H6V20ZM19 20H18V19H19V20ZM3 17V10H21V17H3Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Parking Spaces
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          2
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name calendar"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 4H18V3C18 2.44772 17.5523 2 17 2C16.4477 2 16 2.44772 16 3V4H8V3C8 2.44772 7.55228 2 7 2C6.44772 2 6 2.44772 6 3V4H5C3.34315 4 2 5.34315 2 7V19C2 20.6569 3.34315 22 5 22H19C20.6569 22 22 20.6569 22 19V7C22 5.34315 20.6569 4 19 4ZM20 19C20 19.5523 19.5523 20 19 20H5C4.44772 20 4 19.5523 4 19V10H20V19ZM20 8H4V7C4 6.44772 4.44772 6 5 6H19C19.5523 6 20 6.44772 20 7V8Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Date Signed Up
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          20th September 2023
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name property"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22.9227 9.16484L12.4227 2.16484C12.089 1.94505 11.6565 1.94505 11.3227 2.16484L0.872748 9.16484C0.470624 9.48875 0.381235 10.0656 0.66646 10.496C0.951685 10.9264 1.51772 11.0689 1.97275 10.8248L2.87275 10.1948V21.4948C2.87275 22.0471 3.32046 22.4948 3.87275 22.4948H19.8727C20.425 22.4948 20.8727 22.0471 20.8727 21.4948V10.1948L21.8227 10.8248C21.9838 10.9376 22.1761 10.9971 22.3727 10.9948C22.8113 10.9914 23.1965 10.7027 23.3227 10.2827C23.4489 9.86268 23.2868 9.40943 22.9227 9.16484ZM18.8727 20.4948H4.87275V8.86484L11.8727 4.19484L18.8727 8.86484V20.4948Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text el-text-ellipsis"
        >
          Address
        </div>
        <div
          class="el-text-base el-text-sm el-text-ellipsis"
        >
          I added a textEllipsis prop to this one to show how it works. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-1 el-key-value-list-wrap"
      role="listitem"
    >
      <div
        class="mocked-styled-0 el-key-value-icon-wrap"
      >
        <span
          class="el-intent-primary el-icon"
        >
          <svg
            fill="none"
            height="24"
            role="img"
            title="Icon image with name edit"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 16.1716V21.1716H8L20.59 8.58159C21.3654 7.80151 21.3654 6.54166 20.59 5.76159L18.41 3.58159C17.6299 2.80614 16.3701 2.80614 15.59 3.58159L3 16.1716ZM7.17 19.1716H5V17.0016L13.41 8.58159L15.59 10.7616L7.17 19.1716ZM14.83 7.17159L17 5.00159L19.17 7.17159L17 9.34159L14.83 7.17159Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="el-flex-container el-flex-column"
      >
        <div
          class="el-text-base el-text-xs el-has-grey-text"
        >
          Description
        </div>
        <div
          class="el-text-base el-text-sm"
        >
          This one I just allowed the text to wrap. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`KeyValueList component > should match a snapshot for a grid 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-3 el-grid"
  >
    <div
      class="mocked-styled-5 el-col-half"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name user"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.37998 8.88C8.37998 6.87911 9.99911 5.25551 12 5.25C14.0025 5.25549 15.6245 6.87749 15.63 8.88C15.6272 9.49712 15.5703 10.1128 15.46 10.72C15.3083 12.534 13.8195 13.9453 12 14C10.1805 13.9453 8.69171 12.534 8.54 10.72C8.43267 10.1126 8.37913 9.49685 8.37998 8.88ZM12 12C12.91 12 13.36 11.18 13.49 10.4C13.5796 9.89821 13.6265 9.38972 13.63 8.88C13.678 8.26901 13.3792 7.68276 12.8566 7.36257C12.334 7.04238 11.676 7.04238 11.1534 7.36257C10.6308 7.68276 10.332 8.26901 10.38 8.88C10.3789 9.38949 10.4224 9.8981 10.51 10.4C10.64 11.18 11.09 12 12 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M1 12C1 5.92487 5.92487 1 12 1C14.9174 1 17.7153 2.15893 19.7782 4.22183C21.8411 6.28473 23 9.08262 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 17C9.25 17 7.27 18.27 7 19.49H7.06C10.0671 21.5044 13.9929 21.5044 17 19.49C16.73 18.27 14.75 17 12 17ZM12 15C15.07 15 17.62 16.28 18.55 18.12H18.58C21.9161 14.545 21.7874 8.9604 18.2902 5.54284C14.793 2.12528 9.20695 2.12528 5.70975 5.54284C2.21255 8.9604 2.0839 14.545 5.42 18.12C6.38 16.28 8.93 15 12 15Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Name
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            John Doe
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-5 el-col-half"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name email"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 4H4C2.89543 4 2 4.89543 2 6V18C2 19.1046 2.89543 20 4 20H20C21.1046 20 22 19.1046 22 18V6C22 4.89543 21.1046 4 20 4ZM18.59 6L12 12.59L5.41 6H18.59ZM4 18V7.41L10.59 14C11.3701 14.7754 12.6299 14.7754 13.41 14L20 7.41V18H4Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Email
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            email@example.com
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-5 el-col-half"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name car"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.5 15.5C7.60457 15.5 8.5 14.6046 8.5 13.5C8.5 12.3954 7.60457 11.5 6.5 11.5C5.39543 11.5 4.5 12.3954 4.5 13.5C4.5 14.6046 5.39543 15.5 6.5 15.5Z"
                fill="currentColor"
              />
              <path
                d="M19.5 13.5C19.5 14.6046 18.6046 15.5 17.5 15.5C16.3954 15.5 15.5 14.6046 15.5 13.5C15.5 12.3954 16.3954 11.5 17.5 11.5C18.6046 11.5 19.5 12.3954 19.5 13.5Z"
                fill="currentColor"
              />
              <path
                clip-rule="evenodd"
                d="M21 8H20.72L19.63 4.73C19.0843 3.10029 17.5586 2.0013 15.84 2H8.16C6.43872 2.00274 4.9122 3.10634 4.37 4.74L3.28 8H3C1.89543 8 1 8.89543 1 10V17C1 18.1046 1.89543 19 3 19V20C3 21.1046 3.89543 22 5 22H6C7.10457 22 8 21.1046 8 20V19H16V20C16 21.1046 16.8954 22 18 22H19C20.1046 22 21 21.1046 21 20V19C22.1046 19 23 18.1046 23 17V10C23 8.89543 22.1046 8 21 8ZM6.26 5.37C6.53168 4.55142 7.29752 3.99922 8.16 4H15.84C16.7025 3.99922 17.4683 4.55142 17.74 5.37L18.61 8H5.39L6.26 5.37ZM6 20H5V19H6V20ZM19 20H18V19H19V20ZM3 17V10H21V17H3Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Parking Spaces
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            2
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-5 el-col-half"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name calendar"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 4H18V3C18 2.44772 17.5523 2 17 2C16.4477 2 16 2.44772 16 3V4H8V3C8 2.44772 7.55228 2 7 2C6.44772 2 6 2.44772 6 3V4H5C3.34315 4 2 5.34315 2 7V19C2 20.6569 3.34315 22 5 22H19C20.6569 22 22 20.6569 22 19V7C22 5.34315 20.6569 4 19 4ZM20 19C20 19.5523 19.5523 20 19 20H5C4.44772 20 4 19.5523 4 19V10H20V19ZM20 8H4V7C4 6.44772 4.44772 6 5 6H19C19.5523 6 20 6.44772 20 7V8Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Date Signed Up
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            20th September 2023
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-4 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name property"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22.9227 9.16484L12.4227 2.16484C12.089 1.94505 11.6565 1.94505 11.3227 2.16484L0.872748 9.16484C0.470624 9.48875 0.381235 10.0656 0.66646 10.496C0.951685 10.9264 1.51772 11.0689 1.97275 10.8248L2.87275 10.1948V21.4948C2.87275 22.0471 3.32046 22.4948 3.87275 22.4948H19.8727C20.425 22.4948 20.8727 22.0471 20.8727 21.4948V10.1948L21.8227 10.8248C21.9838 10.9376 22.1761 10.9971 22.3727 10.9948C22.8113 10.9914 23.1965 10.7027 23.3227 10.2827C23.4489 9.86268 23.2868 9.40943 22.9227 9.16484ZM18.8727 20.4948H4.87275V8.86484L11.8727 4.19484L18.8727 8.86484V20.4948Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text el-text-ellipsis"
          >
            Address
          </div>
          <div
            class="el-text-base el-text-sm el-text-ellipsis"
          >
            I added a textEllipsis prop to this one to show how it works. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-4 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-0 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon"
          >
            <svg
              fill="none"
              height="24"
              role="img"
              title="Icon image with name edit"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 16.1716V21.1716H8L20.59 8.58159C21.3654 7.80151 21.3654 6.54166 20.59 5.76159L18.41 3.58159C17.6299 2.80614 16.3701 2.80614 15.59 3.58159L3 16.1716ZM7.17 19.1716H5V17.0016L13.41 8.58159L15.59 10.7616L7.17 19.1716ZM14.83 7.17159L17 5.00159L19.17 7.17159L17 9.34159L14.83 7.17159Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Description
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            This one I just allowed the text to wrap. Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia dolorem laudantium, repellat, aut tenetur enim veritatis debitis nostrum iste fugit quas delectus repellendus iusto ratione esse. Ab rem voluptate veritatis.
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
